src/scripts/generic_coverage.py

```diff
@@ -0,0 +1,200 @@
+    # Retrieve terms under scope in 'ontologies' with IRIs and labels from Ubergraph.
+    # Remove all superclasses of terms on the term list (via subClassOf/ and part_of)
+    mas_dict = get_scope_terms(ontologies, _scope)
+    return clean_up_scope_terms(_term_dict, mas_dict, _scope)
+
+
+def get_scope_terms(ontologies: str, _scope: str) -> Dict[str, str]:
+    # Get multicellular anatomical structure terms with IRIs and labels from Ubergraph
+    sparql.setQuery(get_scope_query(ontologies, _scope))
+    mas_query_response = sparql.queryAndConvert()
+    mas_dict: Dict[str, str] = {}
+    for mas in mas_query_response["results"]["bindings"]:
+        mas_dict.update({mas['mas']['value']: mas['label']['value']})
+    return mas_dict
+
+
+def clean_up_scope_terms(_term_dict: Dict[str, str], mas_dict: Dict[str, str], _scope: str) -> Dict[str, str]:
+    # Remove all superclasses of terms on the scope list (via subClassOf/ and part_of)
+    sparql.setQuery(get_superclass_value_query(list(_term_dict.values()), _scope))
+    super_ret = sparql.queryAndConvert()
+    for super_class in super_ret["results"]["bindings"]:
+        mas_dict.pop(super_class['super']['value'], None)
+    return mas_dict
+
+
+def get_term_leaves(term_list: List[str], _scope: str) -> Dict[str, Dict[str, str]]:
+    # Get terms under, connected with 'subClassOf' relation, given the term list via template file from Ubergraph
+    _term_leaves_dict = {}
+    sparql.setQuery(get_tissue_list_query(term_list, _scope))
+    ret = sparql.queryAndConvert()
+    for row in ret["results"]["bindings"]:
+        if row['organ_label']['value'] not in _term_leaves_dict.keys():
+            _term_leaves_dict.update(
+                {row['organ_label']['value']: {row['tissue']['value']: row['tissue_label']['value']}})
+        else:
+            _term_leaves_dict[row['organ_label']['value']].update(
+                {row['tissue']['value']: row['tissue_label']['value']})
+    return _term_leaves_dict
+
+
+def get_invalid_subclass_list(_term_dict: Dict[str, Dict[str, str]]) -> List[str]:
+    merged_subset = {k: v for key, value in _term_dict.items() for k, v in value.items() if key != value.get(k)}
+    invalid_subclass_list = []
+    for term, term_subset in _term_dict.items():
+        if term in merged_subset.values():
+            invalid_subclass_list.append(term)
+    return invalid_subclass_list
+
+
+def get_scope_query(ontology: str, scope_term: str) -> str:
+    """Returns the SPARQL query to retrieve multicellular anatomical structure terms with IRIs and labels
+
+    Args:
+        ontology (str): Ontology name, can be either "uberon" or "all"
+        scope_term (str): Scope of the slim
+
+    Returns:
+        output (str): Multicellular anatomical structure SPARQL query
+    """
+    # ?mas rdfs:isDefinedBy 'obo:uberon.owl'
+    filter_str = "FILTER(strstarts(str(?mas),'http://purl.obolibrary.org/obo/UBERON_'))" if ontology == 'uberon' else ""
```

Comment:
You can also use the annotation `rdfs:isDefinedBy` here.

Reply:
But `?mas` is already defined by CL in the query.

Reply:
We don't use the ontology parameter anymore, I guess I should have removed this line and the parameter...

Reply:
Rename the variables in the SPARQL query would be good too.